app/models/manageiq/providers/nuage/inventory/collector/target_collection.rb

```diff
@@ -16,13 +16,15 @@ def cloud_networks_floating
   end
 
   def cloud_subnets
-    return [] if references(:cloud_subnets).blank?
+    return [] if references_with_kind(:cloud_subnets, 'L3').blank?
     references(:cloud_subnets).collect { |ems_ref| cloud_subnet(ems_ref) }
     @cloud_subnets_map.values.compact
   end
 
   def l2_cloud_subnets
-    [] # TODO(miha-plesko): targeted refresh for l2_cloud_subnets
+    return [] if references_with_kind(:cloud_subnets, 'L2').blank?
+    references(:cloud_subnets).collect { |ems_ref| l2_cloud_subnet(ems_ref) }
```

Comment:
so why not use ```references_with_kind(:cloud_subnets, 'L2')``` here?

you can do
```
def l2_cloud_subnets
  return [] if (refs = references_with_kind(:cloud_subnets, 'L2')).blank?
  refs.map { |ems_ref| l2_cloud_subnet(ems_ref) }.compact
end
````

same for the other one

Reply:
Thanks for noticing, would cause us headaches in future. I'm not sure why I we don't use `map` with other functions as well instad referencing internal hashes. Will fix in followup PR.